privacyidea/api/lib/prepolicy.py

```diff
@@ -523,11 +526,17 @@ def init_registrationcode_length_contents(request=None, action=None):
                                 user_object=user_object).action_values(unique=True)
         if len(length_pols) == 1:
             request.all_data[ACTION.REGISTRATIONCODE_LENGTH] = list(length_pols)[0]
+            no_length_policy = False
         content_pols = Match.user(g, scope=SCOPE.ENROLL, action=ACTION.REGISTRATIONCODE_CONTENTS,
                                 user_object=user_object).action_values(unique=True)
         if len(content_pols) == 1:
             request.all_data[ACTION.REGISTRATIONCODE_CONTENTS] = list(content_pols)[0]
-
+            no_content_policy = False
+    # if there is no policy, set defaults.
+    if no_length_policy:
```

Comment:
I think, this is an unneccessary boolean, since you could have set the DEFAULT_LENGTH before checking for the policy directly. But I leave it up to you.